### PR TITLE
Provide secondary source location for error messages in standard json.

### LINKED ISSUE #6632: Standard compiler structured error messages do not contain secondary source locations
The function Json::Value formatErrorWithException in StandardComplier.cpp ignores the secondary source location (it is just taken into account for formattedMessage, but not for the structured data).

### PR DESCRIPTION
**Pull request by chriseth**
Fixes https://github.com/ethereum/solidity/issues/6632

### CODE REVIEW

File: libsolidity/interface/StandardCompiler.cpp [hunk 1]
```diff
@@ -70,6 +74,36 @@ Json::Value formatFatalError(string const& _type, string const& _message)
 	return output;
 }
 
+Json::Value formatSourceLocation(SourceLocation const* location)
+{
+	Json::Value sourceLocation;
+	if (location && location->source && !location->source->name().empty())
```
Comment (Marenz) [inline libsolidity/interface/StandardCompiler.cpp:80]: maybe reverse the condition to avoid the block & indent?

Comment (Marenz) [reply libsolidity/interface/StandardCompiler.cpp:80]: Nevermind, I didn't see that you reuse the return

Comment (Marenz) [reply libsolidity/interface/StandardCompiler.cpp:80]: though you could just do `return Json::Value()` I suppose..

File: libsolidity/interface/StandardCompiler.cpp [hunk 2]
```diff
@@ -70,6 +74,36 @@ Json::Value formatFatalError(string const& _type, string const& _message)
...
+		sourceLocation["start"] = location->start;
+		sourceLocation["end"] = location->end;
+	}
+
+	return sourceLocation;
+}
+
+Json::Value formatSecondarySourceLocation(SecondarySourceLocation const* _secondaryLocation)
+{
+	Json::Value secondarySourceLocation;
```
Comment (Marenz) [inline libsolidity/interface/StandardCompiler.cpp]: Maybe here it makes more sense to do `if(!_secondLocation) return Json::Value();`

**Comment by erak** [comment]
Could you add an additional commandline test, which checks that multiple secondary source locations are provided as intended?

File: libsolidity/interface/StandardCompiler.cpp (hunk 1 above)
Comment (chriseth) [reply libsolidity/interface/StandardCompiler.cpp:80]: I would say the function is short enough...

**Comment by chriseth** [comment]
Updated.

**Review by leonardoalt — APPROVED** [review]

**Review by erak — APPROVED** [review]

**Pull request merged**